workers/generic-worker/mounts.go

```diff
@@ -764,13 +747,80 @@ func extract(fsContent FSContent, format string, dir string, taskMount *TaskMoun
 		unarchiver = &archiver.TarZstd{
 			Tar: &archiver.Tar{},
 		}
+	case "tar.lz4":
+		unarchiver = &archiver.TarLz4{
+			Tar: &archiver.Tar{},
+		}
 	default:
 		log.Fatalf("Unsupported format %v", format)
 		return fmt.Errorf("Unsupported archive format %v", format)
 	}
 	return unarchiver.Unarchive(cacheFile, dir)
 }
 
+func decompress(fsContent FSContent, format string, file string, taskMount *TaskMount) error {
+	cacheFile, err := ensureCached(fsContent, taskMount)
+	if err != nil {
+		log.Printf("Could not cache content: %v", err)
+		return err
+	}
+
+	parentDir := filepath.Dir(file)
+	err = MkdirAll(taskMount, parentDir, 0700)
+	// this could be a user error, if someone supplies an invalid path, so let's not
+	// panic, but make this a task failure
+	if err != nil {
+		return err
+	}
+
+	var d archiver.Decompressor
+	switch format {
+	case "bz2":
+		d = &archiver.Bz2{}
+	case "gz":
+		d = &archiver.Gz{}
+	case "lz4":
+		d = &archiver.Lz4{}
+	case "xz":
+		d = &archiver.Xz{}
+	case "zst":
+		d = &archiver.Zstd{}
+	case "":
+		// No compression, just copy file.
+		// Let's copy rather than move, since we want to be totally sure that the
+		// task can't modify the contents, and setting as read-only is not enough -
+		// the user could change the rights and then modify it.
+		taskMount.Infof("Copying %v to %v", cacheFile, file)
+		err = copyFileContents(cacheFile, file)
+		if err != nil {
+			// this could be a system error, but it can also be that e.g. the task
+			// specified an invalid path, so resolve as malformed payload rather
+			// than panic
+			taskMount.Errorf("Not able to mount content from %v at path %v", fsContent.String(), file)
+			return err
+		}
+		return nil
+	default:
+		log.Fatalf("Unsupported format %v", format)
```

Comment:
I think this was originally added [here](https://github.com/taskcluster/taskcluster/pull/5093/files#diff-0c9721edee76ea6b023e15639a4e25f9d51080ccedc80b6d19bdb0e1593f7b16R686) and I hadn't spotted that it was using `log.Fatalf`. But that is bad, because it will cause worker to immediately exit without resolving the task. Instead this should either be a `panic` (that would be recovered from, so task gets resolved as exception and logs are uploaded) or it should cause task to resolve as malformed payload. Note this can only happen if schema becomes out of sync with the switch statement, i.e. there is a format in the schema that is missing from the switch statement. But if that happens, we would like task logs, so we should fix this.